pkg/bsql/query.go

```diff
@@ -201,6 +201,67 @@ func (s *SQLSyncer) setupPagination(ctx context.Context, pToken *pagination.Toke
 	return ret, nil
 }
 
+func (s *SQLSyncer) prepareProvisioningQuery(ctx context.Context, query string, vars map[string]any) (string, []interface{}, error) {
+	var qArgs []interface{}
+
+	var parseErr error
+	updatedQuery := queryOptRegex.ReplaceAllStringFunc(query, func(token string) string {
+		key := strings.ToLower(strings.TrimSuffix(strings.TrimPrefix(token, "?<"), ">"))
+
+		if v, ok := vars[key]; ok {
+			qArgs = append(qArgs, v)
+		} else {
+			parseErr = errors.Join(parseErr, fmt.Errorf("unknown token %s", token))
+			return token
+		}
+
+		return s.getNextPlaceholder(qArgs)
+	})
+	if parseErr != nil {
+		return "", nil, parseErr
+	}
+	return updatedQuery, qArgs, nil
+}
+
+func (s *SQLSyncer) runProvisioningQueries(ctx context.Context, queries []string, vars map[string]any) error {
+	l := ctxzap.Extract(ctx)
+
+	tx, err := s.db.Begin()
+	if err != nil {
+		return err
+	}
+	defer func() {
+		if err := tx.Rollback(); err != nil {
```

Comment:
Will tx.Rollback() error if the transaction has already been committed? If so, would that cause this function to return an error on success?

Reply:
I believe that the db driver handles this for us, but made it more explicit in [fc3935d](https://github.com/ConductorOne/baton-sql/pull/31/commits/fc3935dbae8e51aa0e228c785b23211e7f9c97d1) in case we use a driver that doesn't handle it appropriately.